python/tank/util/metrics.py

```diff
@@ -345,79 +348,164 @@ def data(self):
+                Trial Signup
+                User Signup
+                App
+                Onboarding
+                Account Settings
+                Projects
+                Entities
+                Tasks
+                People
+                Media
+                Notes
+                Subscribe
+
+
+        :param: str event_name: A short description of the performed action.
+            The complete list can be found in the 'Shotgun Event Taxonomy' document.
+            Below are a few examples:
+                'Viewed Login Page'
+                'Logged In'
+                'Created Project'
+                'Toggled Project Favorite'
+                'Edited Task Status'
+                'Read Inbox Item'
         """
-        super(UserActivityMetric, self).__init__({
-            "type": "user_activity",
-            "module": module,
-            "action": action,
+
+        if event_group:
+            if not isinstance(event_group, str):
+                raise TypeError("The `event_group` parameter must be None or a str")
+
+        if event_name:
+            if not isinstance(event_name, str):
+                raise TypeError("The `event_name` parameter must be None or a str")
+
+        # TODO: go for silent cast to 'str' or raise TypeError ?
+        super(EventMetric, self).__init__({
+            "event_group": str(event_group),
+            "event_name": str(event_name)
         })
 
-    def __repr__(self):
-        """Official str representation of the user activity metric."""
-        return "%s.%s" % (self._data["module"], self._data["action"])
+        # Initializing the event property dictionary with a default event_type
+        # somehow duplicating what's being done in Shogun (The server)
+        self._data["event_property"] = {"event_type" : "event"}
 
+    def add_event_property(self, name, value):
+        # TODO: add check or warning about possibly conflicting event properties used in Amplitude?
+        # TODO: (e.g.: 'city', 'ip_address', 'device_manufacturer', 'device_type' just to name a few )
+        self._data["event_property"][name] = value
 
-class UserAttributeMetric(ToolkitMetric):
-    """Convenience class for a user attribute metric."""
+    def add_user_info_properties(self):
+        """
+        Helper method adding a number of user information metrics such as:
+        - Approx. location
+        - City
+        - Country
+        - Language
+
+        The method takes care of gathering relevant information and properly
+        formatting them for submission to a Shotgun site
+        """
 
-    def __init__(self, attr_name, attr_value):
-        """Initialize the metric with the attribute name and value.
-        
-        :param str attr_name: Name of the attribute.
-        :param str attr_value: The value of the attribute.
+        # TODO: find a descent library and implement method
+        pass
 
+    def add_system_info_properties(self):
```

Comment:
See my comments above. I think we should make this private and have it auto-populate for each metric created. See the [platform module](https://docs.python.org/2/library/platform.html#module-platform) for access to this info.